syntax = "proto3";

message Null {}

message Task {
  int32 id = 1;
  string description = 2;
  int32 course_id = 3;
  string login = 4;
  int32 score = 5;
}

message TaskList {
  repeated Task items = 1;
}

message GetTaskArg {
  int32 id = 1;
  int32 course_id = 2;
  string login = 3;
}

message GetAllTasksArg {
  int32 course_id = 1;
  string login = 2;
}

service TaskService {
  rpc AddTask(Task) returns (Null) {}
  rpc GetTask(GetTaskArg) returns (Task) {}
  rpc GetAllTasks(GetAllTasksArg) returns (TaskList) {}
}
